File: server/src/auth/auth.controller.ts
import { Body, Controller, Post, UsePipes } from "@nestjs/common";
import {CreateUserDto} from "../users/dto/create-user.dto";
import {AuthService} from "./auth.service";
import { ValidationPipe } from "../pipes/validation.pipe";
import { ApiBody, ApiConsumes, ApiOkResponse, ApiOperation, ApiUnauthorizedResponse } from "@nestjs/swagger";
import { LoginUserDto } from "../users/dto/login-user.dto";
import { LoggedInTokenDto } from "../users/dto/logged-in-token.dto";

@Controller('auth')
export class AuthController {

    constructor(private authService: AuthService){}

    @ApiOperation({summary: "Log in"})
    @ApiConsumes("application/json")
    @ApiBody({type: LoginUserDto})
    @ApiUnauthorizedResponse({description: "Incorrect email or password"})
    @ApiOkResponse({description: "JWT token", type: LoggedInTokenDto})
    @Post('/login')
    login(@Body() userDto: CreateUserDto){
        return this.authService.login(userDto);
    }

    @ApiOperation({summary: "Sign in"})
    @ApiConsumes("application/json")
    @ApiBody({type: CreateUserDto})
    @Post('/reg')
    @UsePipes(ValidationPipe)
    registration(@Body() userDto: CreateUserDto){
        return this.authService.registration(userDto);
    }
}

File: server-java/src/main/java/com/muz/controller/AuthController.java
package com.muz.controller;

import com.muz.dto.CreateUserDto;
import com.muz.dto.LoginUserDto;
import com.muz.dto.TokenResponseDto;
import com.muz.service.AuthService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/auth")
@RequiredArgsConstructor
@Tag(name = "Auth")
public class AuthController {

    private final AuthService authService;

    @PostMapping("/login")
    @Operation(summary = "Login with email and password")
    public TokenResponseDto login(@RequestBody LoginUserDto dto) {
        return authService.login(dto);
    }

    @PostMapping("/reg")
    @Operation(summary = "Register a new user")
    public TokenResponseDto register(@Valid @RequestBody CreateUserDto dto) {
        return authService.registration(dto);
    }
}
